config/main.py

```diff
@@ -570,6 +570,82 @@ def is_ipaddress(val):
         return False
     return True
 
+#
+# Check if an interface_name is in a vlan
+#
+def  interface_is_in_vlan(vlan_member_table, interface_name):
+
+    for k,v in vlan_member_table:
+        if v == interface_name:
+            return True
+
+    return False
+
+
+#
+# Check if port is already configured as mirror destination port
+#
+def interface_is_mirror_dst_port(config_db, interface_name):
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+    for k,v in mirror_table.items():
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+#
+# Check if port is already configured with mirror config
+#
+def interface_has_mirror_config(mirror_table, interface_name):
+    for k,v in mirror_table.items():
+        if 'src_port' in v and v['src_port'] == interface_name:
+            return True
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+#
+# Check if SPAN mirror-session config is valid.
+#
+def validate_mirror_session_config(config_db, session_name, dst_port, src_port, direction):
+    if len(config_db.get_entry('MIRROR_SESSION', session_name)) != 0:
+        click.echo("Error: {} already exists".format(session_name))
+        return False
+
+    vlan_member_table = config_db.get_table('VLAN_MEMBER')
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+
+    if dst_port is not None:
+        if interface_name_is_valid(dst_port) is False:
+            click.echo("Error: Destination Interface {} is invalid".format(dst_port))
+            return False
+
+        if interface_is_in_vlan(vlan_member_table, dst_port):
+            click.echo("Error: Destination Interface {} has vlan config".format(dst_port))
+            return False
+
+        if interface_has_mirror_config(mirror_table, dst_port):
```

Comment:
Not sure if I understand this correctly, can a single dst port cannot be a monitor (mirror-to) port
for multiple mirror sessions?
For example
case 1: mirror rx from Ethernet0 to Etherent4
case 2: mirror rx&tx from Ethernet2 to Etherent4
with the above check, case 2 will fail?

Reply:
Yes. This is not supported. We allow only one mirror session per port, The session can have any number of ports.